File: SistComp/api/src/main/java/com/example/api/models/AlimentoEstocado.java
package com.example.api.models;

import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.*;
import lombok.*;


@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Entity
@Table(name = "Tcc_Alimentos")
public class AlimentoEstocado {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idEstoque;
    
    @ManyToOne
    @JoinColumn(name = "idAlimento", nullable = false)
    private Alimento alimentoASerEstocado;

    @Column(nullable = true)
    private String especificacoes;

    @Column(nullable = false)
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate validade;

    @ManyToOne
    @JoinColumn(name = "idCliente", nullable = false)
    private Cliente cliente;

    @Column(nullable = false)
    private int quantidadeEstoque;

}

File: SistComp/NovaVersao/app/src/main/java/com/example/appcomida/dataclass/AlimentoEstocado.kt
package com.example.appcomida.dataclass

import java.time.LocalDate

data class AlimentoEstocado(
    val alimentoId: Int? = null,
    val alimentoASerEstocado: Alimento? = null,
    val especificacoes: String? = null,
    val validade: LocalDate,
    val clienteId: Int = 0
)

